py/selenium/webdriver/firefox/options.py

```diff
@@ -87,14 +86,10 @@ def to_capabilities(self):
 
             returns a dictionary with everything
         """
-        firefox = DesiredCapabilities.FIREFOX.copy()
-        firefox_options = self._firefox_options
+        capabilities = self._firefox_options
         if self.binary_location:
-            firefox_options["binary"] = self.binary_location
+            capabilities["binary"] = self.binary_location
         if self._profile:
-            firefox_options["profile"] = self._profile.encoded
-        firefox_options["args"] = self.arguments
-
-        firefox["requiredCapabilities"] = firefox_options
-
-        return firefox
+            capabilities["firefox_profile"] = self._profile.encoded
```

Comment:
GeckoDriver is expecting `firefox_profile`, but I'm not sure if this should be fixed here or there.. is there some sort of reference document for capabillties?